src/plugins/procdump/private.h

```diff
@@ -146,30 +150,65 @@ enum
     MMPTE_PROTOTYPE,
 };
 
+#define MM_GUARD_PAGE         0x10
+
+// Assume that valid page is accessible
 static bool IS_MMPTE_VALID(uint64_t mmpte)
 {
     return VMI_GET_BIT(mmpte, 0);
 }
 
-static bool IS_MMPTE_TRANSITION(uint64_t mmpte)
+// Assume that MM_GUARD_PAGE shoulbe reset to page been accessible
+static bool IS_MMPTE_ACCESSIBLE(uint64_t protection)
 {
-    return !IS_MMPTE_VALID(mmpte) && VMI_GET_BIT(mmpte, 11);
+    return (protection & 0x7) && !(protection & MM_GUARD_PAGE);
 }
 
+/*
+ * Assume that either the MMPTE is valid or prototype or transition or software
+ */
+
 static bool IS_MMPTE_PROTOTYPE(uint64_t mmpte)
 {
-    return !IS_MMPTE_VALID(mmpte) && !IS_MMPTE_TRANSITION(mmpte) &&
-           VMI_GET_BIT(mmpte, 10);
+    return VMI_GET_BIT(mmpte, 10);
+}
+
+static bool IS_MMPTE_TRANSITION(uint64_t mmpte)
+{
+    return VMI_GET_BIT(mmpte, 11);
+}
+
+static bool IS_MMPTE_PROTOTYPE_ACCESSIBLE(uint64_t mmpte)
+{
+    auto is_accessible = IS_MMPTE_ACCESSIBLE((mmpte >> 11));
+    return VMI_GET_BIT(mmpte, 10) &&
```

Comment:
Could you use named constant instead of `11` or provide a comment?

Also, below we do have:
```
auto is_accessible = IS_MMPTE_ACCESSIBLE((mmpte >> 5));
```

and it's not clear at the very first glance what these `>> 11` and `>> 5` shifts are doing without additional investigation of the layout of MMPTE

Reply:
@icedevml done!